account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/core/CredentialBundle.java

```diff
@@ -0,0 +1,125 @@
+package com.alchemy.aa.core;
+
+import com.alchemy.aa.core.exceptions.StamperNotInitializedException;
+import com.google.common.primitives.Bytes;
+import com.google.crypto.tink.subtle.EllipticCurves;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.Security;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECPoint;
+import org.bitcoinj.base.Base58;
+import org.bouncycastle.jce.ECNamedCurveTable;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jce.spec.ECPublicKeySpec;
+
+public record CredentialBundle(
+    byte[] bundlePrivateKey,
+    byte[] bundlePublicKey
+) {
+
+    /// Private key deciphered from bundle.
+
+    /// Public key from bundle.
+
+    /**
+     * Injects a credential bundle (Base58-encoded) after decrypting.
+     *
+     * @param bundle
+     *            Base58-encoded credential bundle
+     *
+     * @throws GeneralSecurityException
+     *             if initTek is never called.
+     */
+    public static CredentialBundle injectCredentialBundle(String bundle, TekManager tekManager)
```

Comment:
nit: Since this mostly a factory for CredentialBundle, I would rename this to `fromBundle` or `fromEncryptedBundle`?